server/ai_http.go

```diff
@@ -140,9 +138,86 @@ func (h *lphttp) StartLiveVideoToVideo() http.Handler {
 		}()
 
 		// TODO subscribe to the subscribeUrl for output monitoring
+        // ctx := clog.AddVal(r.Context(), clog.ClientIP, remoteAddr)
+
+        var req worker.GenLiveVideoToVideoJSONRequestBody
+        if err := jsonDecoder[worker.GenLiveVideoToVideoJSONRequestBody](&req, r); err != nil {
+            respondWithError(w, err.Error(), http.StatusBadRequest)
+            return
+        }
+
+		appendHostname := func(urlPath string) (*url.URL, error) {
+			if urlPath == "" {
+				return nil, fmt.Errorf("invalid url from orch")
+			}
+			pu, err := url.Parse(urlPath)
+			if err != nil {
+				return nil, err
+			}
+			if pu.Hostname() != "" {
+				// url has a hostname already so use it
+				return pu, nil
+			}
+			// no hostname, so append one
+			_, port, err := net.SplitHostPort(r.Host)
```

Comment:
this looks sus - the request `r` comes from the gateway. This `appendHostname` dance probably needs to be done on the worker before sending the request down to the runner

Reply:
Totally agree, I've moved this to `ai_process.go` where it uses `sess.transcoder()`. I will do tests to ensure this works with all different IP types